processing/src/main/java/io/druid/query/groupby/GroupByQueryHelper.java

```diff
@@ -64,12 +65,15 @@
   )
   {
     final GroupByQueryConfig querySpecificConfig = config.withOverrides(query);
-    final QueryGranularity gran = query.getGranularity();
+    final Granularity gran = query.getGranularity();
     final long timeStart = query.getIntervals().get(0).getStartMillis();
 
     // use gran.iterable instead of gran.truncate so that
     // AllGranularity returns timeStart instead of Long.MIN_VALUE
-    final long granTimeStart = gran.iterable(timeStart, timeStart + 1).iterator().next();
+    long granTimeStart = timeStart;
+    if (!(Granularity.ALL.equals(gran))) {
+      granTimeStart = gran.getIterable(new Interval(timeStart, timeStart + 1)).iterator().next().getStartMillis();
```

Comment:
It seems to me that strange construction `gran.iterable(timeStart, timeStart + 1).iterator().next()` was used only to workaround some undesirable behaviour of AllGranularity, when just gran.truncate() is called instead. Now you explicitly excluded AllGranularity using if condition (that is good), so for all other granularities inside the if just granTimeStart = `gran.truncate(timeStart)` could be used.